spec/requests/bookmarks_spec.rb

```diff
@@ -0,0 +1,249 @@
+require 'rails_helper'
+
+describe 'POST /bookmarks' do
+  # create a user before the test scenarios are run
+  let!(:user) { User.create(username: 'soulchild', authentication_token: 'abcdef') }
+
+  # pass the user username and authentication to the header
+  scenario 'valid bookmark attributes' do
```

Comment:
What is the purpose of this and all other occurrences of it?

Reply:
Written "let" for creating a user before the test scenarios are run.
Written "scenario" to pass the user username and authentication to the header.